docs/code-organisation-guidelines.md

```diff
@@ -0,0 +1,65 @@
+# Code Organisation Guidelines
```

Comment:
This isn't mentioned in the proposal, but something that *frequently* gets in the way of my navigation of the codebase is the prevalence of `autoexporter`. I'll see a line of code like

```haskell
import Foo.Bar (Baz(..))
```

and navigate to `Foo.Bar` only to find an `autoexporter` pragma, and then I need to do `grep "\(data\|newtype\|type\) Baz" **/*.hs` to actually find the thing.

I should probably just get a `tags` setup going, but I generally prefer solutions that require minimal tooling :)

Reply:
Yes, I too find that the autoexporter make navigating the code more difficult.

Reply:
> I should probably just get a tags setup going

You should definitely get a tags setup going with or without autoexporter. Tags are awesome.

Reply:
Wholeheartedly agree with the sentiment here -- I did fall victim of the same trap myself in the past 😁 Not only that, but this also slows us down then using `weeder` as the tool isn't capable (at least at the time of writing) to track down `autoexporter` as being used just merely from the use of the GHC pragma. One might say it's a shortcoming of `weeder` itself, but personally I think `autoexporter` doesn't buy us enough to compensate the difficulty in navigating the codebase. I also vote for its removal 😉 

> Tags are awesome.

Yup, they are 😀